### DISCUSSION TITLE
Limiting the number of results in GraphQL

### DISCUSSION THREAD

**Discussion by PeerMohamedMydeenA** [Azure/data-api-builder#2093] [Q&A]
Is there a setting/option in the DAB config to set/limit the maximum number of results on the graphql endpoint ?

**Comment by severussundar** [comment]
Hi @PeerMohamedMydeenA ,

There is no property in the config file to limit the number of results for a GraphQL response. 

However, this can be achieved by using the `first` parameter directly in the query.

```graphql
{
    books(first: n){
        items{
            ...
        }
     }
}
```
This will result in the response having `n` items (or lesser if the total number of items present in the table is lesser than `n`) . 

For reference, here is the [link](https://learn.microsoft.com/en-us/azure/data-api-builder/graphql#first-and-after) to the docs that explains `first` parameter in detail.

**Reply by PeerMohamedMydeenA** [reply]
Hi @severussundar, 

Sorry for the ambiguous question. Yes, the query can set this limit, however, as an owner of the GraphQL endpoint, I want to make sure the endpoint is not abused or prevent from any DDOS attacks where a user always querying larger size of results every time. I know there can be a rate limiter to prevent frequent queries by a particular client or a timeout to prevent larger queries, but wanted to restrict the query size on the server side.

Speaking of timeout, do we have a DAB config to set a timeout ? I see the query times out at 30 seconds and wanted to see if we can set it to a smaller value.  

Hope my ask is bit clear now.

**Reply by severussundar** [reply]
Hey @PeerMohamedMydeenA , thank you for sharing additional information.

At the moment, we don't have a properties in the config file to a) Limit the size of the graphQL API response b) Making timeouts configurable.

I've created this issue ~ [#2098 ](https://github.com/Azure/data-api-builder/issues/2098) to evaluate this.
